### PR TITLE
fix Content-Type bug

### PR DESCRIPTION
**Pull request by fffguo**
header use Content-Type，program error taking content-type lead to undefined bug

### CODE REVIEW

**Comment by fffguo** [comment]
image

File: index.js [hunk 1]
```diff
@@ -43,8 +43,12 @@ function curlString(
...
 }
 
-function getHeader(options, headerKeyName) {
-  return options.headers[headerKeyName];
+function getLowHeaders(headers) {
+  let lowHeaders = {}
+  Object.keys(headers).forEach(function (key) {
+    lowHeaders[key.toLowerCase()] = headers[key]
+  })
+  return lowHeaders
```
Comment (zvakanaka) [inline index.js]:
```suggestion
function getHeader(options, headerKeyName) {
  // return header that matches case, but if not found fall back to header that does not match case
  return options.headers[headerKeyName] || options.headers[Object.keys(options.headers).find(key => key.toLowerCase() === headerKeyName.toLowerCase())]
```

File: index.js [hunk 2]
```diff
@@ -67,10 +71,11 @@ function bodyToDataString(options, curlStringOptions) {
     // fall back to original body if it could not be parsed as JSON
     parsedData = options.body;
   }
+  const lowHeaders = getLowHeaders(options.headers)
```
Comment (zvakanaka) [inline index.js]:
```suggestion
```

File: index.js [hunk 3]
```diff
@@ -67,10 +71,11 @@ function bodyToDataString(options, curlStringOptions) {
...
 
   // return an ampersand delimited string
   if (hasHeader(options, 'content-type') &&
-      getHeader(options, 'content-type').toLowerCase() === 'application/x-www-form-urlencoded') {
+      lowHeaders['content-type'].toLowerCase() === 'application/x-www-form-urlencoded') {
```
Comment (zvakanaka) [inline index.js]:
```suggestion
      getHeader(options, 'content-type').toLowerCase() === 'application/x-www-form-urlencoded') {
```

**Review by zvakanaka — COMMENTED** [review]
Your changes work well, but I've made some suggestions to match the style of this project. Once those are committed, we can merge this. Thanks for finding this bug and thank you for finding out how to fix it!

**Review by zvakanaka — APPROVED** [review]
Thank you!

**Pull request merged**